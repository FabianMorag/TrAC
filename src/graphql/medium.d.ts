export type Program = import("../../api/entities/program").Program;
export type User = import("../../api/entities/user").User;
export type Student = import("../../api/entities/student").Student;
export type AuthResult = import("../../api/entities/auth/auth").AuthResult;
