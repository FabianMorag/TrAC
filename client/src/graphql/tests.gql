mutation LoginTest($email: EmailAddress!, $password: String!) {
  login(email: $email, password: $password) {
    user {
      email
    }
    token
    error
  }
}

mutation UnlockTest(
  $email: EmailAddress!
  $password: String!
  $unlockKey: String!
) {
  unlock(email: $email, password: $password, unlockKey: $unlockKey) {
    user {
      email
    }
    token
    error
  }
}

query CurrentUserTest {
  currentUser {
    user {
      email
    }
    token
    error
  }
}
