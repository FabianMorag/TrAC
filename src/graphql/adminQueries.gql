fragment UserAdminFragment on User {
  email
  name
  tries
  type
  student_id
  config
  locked
  programs {
    id
  }
}

query allUsersAdmin {
  users {
    ...UserAdminFragment
  }
}

query allProgramsAdmin {
  programs {
    id
  }
}

mutation addUsersProgramsAdmin($user_programs: [UserProgram!]!) {
  addUsersPrograms(user_programs: $user_programs) {
    ...UserAdminFragment
  }
}

mutation updateUserProgramsAdmin($userPrograms: UpdateUserPrograms!) {
  updateUserPrograms(userPrograms: $userPrograms) {
    ...UserAdminFragment
  }
}

mutation upsertUsersAdmin($users: [UpsertedUser!]!) {
  upsertUsers(users: $users) {
    ...UserAdminFragment
  }
}

mutation deleteUserAdmin($email: EmailAddress!) {
  deleteUser(email: $email) {
    ...UserAdminFragment
  }
}

mutation lockMailUserAdmin($email: EmailAddress!) {
  lockMailUser(email: $email) {
    mailResult
    users {
      ...UserAdminFragment
    }
  }
}

mutation mailAllLockedUsersAdmin {
  mailAllLockedUsers
}

mutation editConfigAdmin($name: String!, $value: String!) {
  editConfig(name: $name, value: $value)
}

query userPersistencesAdmin($user: String!) {
  userPersistences(user: $user) {
    key
    data
    timestamp
  }
}

mutation resetPersistenceAdmin($user: String!) {
  resetPersistence(user: $user)
}

mutation resetDataLoadersCacheAdmin {
  resetDataLoadersCache
}

mutation feedbackResultsCsvAdmin {
  feedbackResultsCsv
}

query feedbackResultsAdmin($user_ids: [String!]) {
  feedbackResults(user_ids: $user_ids) {
    user {
      email
    }
    form {
      name
    }
    answers {
      answer
      question {
        question
        type
        options {
          text
          value
        }
      }
    }
    timestamp
  }
}

query trackInfo($minDate: DateTime!, $maxDate: DateTime!) {
  trackInfo(minDate: $minDate, maxDate: $maxDate) {
    id
    user_id
    data
    app_id
    datetime
  }
}
