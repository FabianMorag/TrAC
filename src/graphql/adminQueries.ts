import gql, { DocumentNode } from "graphql-tag-ts";

import { UserType } from "../../constants";
import { IfImplements } from "../../typings/utils";
import { Program, User } from "./medium";

type IAllUsersAdmin = IfImplements<
  {
    email: string;
    name: string;
    tries: number;
    type: UserType;
    rut_id?: string;
    show_dropout: boolean;
    locked: boolean;
    programs: { id: string }[];
  },
  User
>[];

const AllUsersAdminFragment = gql`
  fragment allUsersAdminFragment on User {
    email
    name
    tries
    type
    rut_id
    show_dropout
    locked
    programs {
      id
    }
  }
`;

export const ALL_USERS_ADMIN: DocumentNode<{
  users: IAllUsersAdmin;
}> = gql`
  query {
    users {
      ...allUsersAdminFragment
    }
  }
  ${AllUsersAdminFragment}
`;

export const ALL_PROGRAMS_ADMIN: DocumentNode<{
  programs: IfImplements<{ id: string }, Program>[];
}> = gql`
  query {
    programs {
      id
    }
  }
`;

export const ADD_USERS_PROGRAMS_ADMIN: DocumentNode<
  {
    addUsersPrograms: IAllUsersAdmin[];
  },
  {
    user_programs: {
      email: string;
      program: string;
    }[];
  }
> = gql`
  mutation($user_programs: [UserProgram!]!) {
    addUsersPrograms(user_programs: $user_programs) {
      ...allUsersAdminFragment
    }
  }
  ${AllUsersAdminFragment}
`;

export const UPDATE_USER_PROGRAMS_ADMIN: DocumentNode<
  {
    updateUserPrograms: IAllUsersAdmin;
  },
  {
    update_user: {
      email: string;
      programs: string[];
      oldPrograms: string[];
    };
  }
> = gql`
  mutation($update_user: UpdateUserPrograms!) {
    updateUserPrograms(userPrograms: $update_user) {
      ...allUsersAdminFragment
    }
  }
  ${AllUsersAdminFragment}
`;

export const UPSERT_USERS_ADMIN: DocumentNode<
  {
    upsertUsers: IAllUsersAdmin;
  },
  {
    users: {
      oldEmail?: string;
      email: string;
      name?: string;
      type?: UserType;
      tries?: number;
      rut_id?: string;
      show_dropout?: boolean;
      locked?: boolean;
    }[];
  }
> = gql`
  mutation($users: [UpsertedUser!]!) {
    upsertUsers(users: $users) {
      ...allUsersAdminFragment
    }
  }
  ${AllUsersAdminFragment}
`;

export const DELETE_USER_ADMIN: DocumentNode<
  {
    deleteUser: IAllUsersAdmin;
  },
  {
    email: string;
  }
> = gql`
  mutation($email: EmailAddress!) {
    deleteUser(email: $email) {
      ...allUsersAdminFragment
    }
  }
  ${AllUsersAdminFragment}
`;

export const LOCK_MAIL_USER_ADMIN: DocumentNode<
  {
    lockMailUser: {
      mailResult: Record<string, any>;
      users: IAllUsersAdmin;
    };
  },
  { email: string }
> = gql`
  mutation($email: EmailAddress!) {
    lockMailUser(email: $email) {
      mailResult
      users {
        ...allUsersAdminFragment
      }
    }
  }
  ${AllUsersAdminFragment}
`;

export const MAIL_LOCKED_USERS_ADMIN: DocumentNode<{
  mailAllLockedUsers: Record<string, any>[];
}> = gql`
  mutation {
    mailAllLockedUsers
  }
`;
