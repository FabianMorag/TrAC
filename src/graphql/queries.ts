import gql, { DocumentNode } from "graphql-tag-ts";

import { AuthResult } from "../../api/entities/auth";
import { baseConfig, StateCourse, TermType, UserType } from "../../constants";
import { IfImplements } from "../../typings/utils";
import { Program, Student, User } from "./medium";

export const UserFragment = gql`
  fragment UserFragment on User {
    email
    name
    admin
    type
    show_dropout
    show_student_list
  }
`;

export const LOGIN: DocumentNode<
  {
    login: {
      user?: {
        email: string;
        name: string;
        admin: boolean;
        type: UserType;
        show_dropout: boolean;
        show_student_list: boolean;
      };
      error?: string;
    };
  },
  {
    email: string;
    password: string;
  }
> = gql`
  mutation($email: EmailAddress!, $password: String!) {
    login(email: $email, password: $password) {
      user {
        ...UserFragment
      }
      error
    }
  }
  ${UserFragment}
`;

export const CURRENT_USER: DocumentNode<{
  currentUser?: IfImplements<
    {
      user?: {
        email: string;
        name: string;
        admin: boolean;
        type: UserType;
        show_dropout: boolean;
        show_student_list: boolean;
      };
    },
    AuthResult
  >;
}> = gql`
  query {
    currentUser {
      user {
        ...UserFragment
      }
    }
  }
  ${UserFragment}
`;

export const UNLOCK: DocumentNode<
  {
    unlock: {
      user?: IfImplements<
        {
          email: string;
          name: string;
          admin: boolean;
          type: UserType;
          show_dropout: boolean;
          show_student_list: boolean;
        },
        User
      >;
      error?: string;
    };
  },
  {
    email: string;
    password: string;
    unlockKey: string;
  }
> = gql`
  mutation($email: EmailAddress!, $password: String!, $unlockKey: String!) {
    unlock(email: $email, password: $password, unlockKey: $unlockKey) {
      user {
        ...UserFragment
      }
      error
    }
  }
  ${UserFragment}
`;

export const LOGOUT: DocumentNode<{
  logout: boolean;
}> = gql`
  mutation {
    logout
  }
`;

export const SEARCH_PROGRAM: DocumentNode<
  {
    program: IfImplements<
      {
        id: string;
        name: string;
        desc: string;
        active: boolean;
        curriculums: {
          id: string;
          semesters: {
            id: number;
            courses: {
              code: string;
              name: string;
              credits: { label: string; value: number }[];
              mention: string;
              flow: { code: string }[];
              requisites: {
                code: string;
              }[];
              historicalDistribution: {
                label: string;
                value: number;
              }[];
              bandColors: { min: number; max: number; color: string }[];
            }[];
          }[];
        }[];
      },
      Program
    > | null;
  },
  { program_id?: string; student_id?: string }
> = gql`
  mutation($program_id: String, $student_id: String) {
    program(id: $program_id, student_id: $student_id) {
      id
      name
      desc
      active
      curriculums {
        id
        semesters {
          id
          courses {
            code
            name
            credits {
              label
              value
            }
            mention
            flow {
              code
            }
            requisites {
              code
            }
            historicalDistribution {
              label
              value
            }
            bandColors {
              min
              max
              color
            }
          }
        }
      }
    }
  }
`;

export const SEARCH_STUDENT: DocumentNode<
  {
    student?: IfImplements<
      {
        id: string;
        programs: {
          id: string;
          name: string;
        }[];
        curriculums: string[];
        start_year: number;
        mention: string;
        terms: Array<{
          id: number;
          student_id: string;
          year: number;
          term: TermType;
          situation: string;
          semestral_grade: number;
          cumulated_grade: number;
          program_grade: number;
          takenCourses: Array<{
            id: number;
            code: string;
            name: string;
            registration: string;
            grade: number;
            state: StateCourse;
            parallelGroup: number;
            currentDistribution: Array<{
              label: string;
              value: number;
            }>;
          }>;
        }>;
        dropout?: {
          prob_dropout: number;
          model_accuracy: number;
          active: boolean;
        };
      },
      Student
    >;
  },
  {
    student_id: string;
    program_id: string;
  }
> = gql`
  mutation($student_id: String!, $program_id: String!) {
    student(student_id: $student_id, program_id: $program_id) {
      id
      programs {
        id
        name
      }
      curriculums
      start_year
      mention
      terms {
        id
        student_id
        year
        term
        situation
        semestral_grade
        cumulated_grade
        program_grade
        takenCourses {
          id
          code
          name
          registration
          grade
          state
          parallelGroup
          currentDistribution {
            label
            value
          }
        }
      }
      dropout {
        prob_dropout
        model_accuracy
        active
      }
    }
  }
`;

export const MY_PROGRAMS: DocumentNode<{
  myPrograms: IfImplements<{ id: string; name: string }, Program>[];
}> = gql`
  query {
    myPrograms {
      id
      name
    }
  }
`;

export const trackMutation: DocumentNode<
  never,
  {
    datetime_client: Date;
    data: string;
  }
> = gql`
  mutation($data: String!, $datetime_client: DateTime!) {
    track(data: $data, datetime_client: $datetime_client)
  }
`;

export const configQuery: DocumentNode<{
  config: typeof baseConfig;
}> = gql`
  query {
    config
  }
`;
