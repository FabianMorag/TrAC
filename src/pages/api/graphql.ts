import connect, { SimpleHandleFunction } from "connect";
import { createProxyMiddleware } from "http-proxy-middleware";
import { toInteger } from "lodash";

const app = connect();

const silentWarn = (message: any) => {
  if (
    message !==
    "API resolved without sending a response for /api/graphql, this may result in stalled requests."
  ) {
    console.error(message);
  }
};
console.warn = silentWarn;

const port = process?.env?.API_PORT ? toInteger(process.env.API_PORT) : 4000;

app.use(
  createProxyMiddleware(`http://localhost:${port}`) as SimpleHandleFunction
);

export const config = {
  api: {
    bodyParser: false,
  },
};

export default app;
