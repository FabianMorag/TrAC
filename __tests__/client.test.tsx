import "@testing-library/jest-dom/extend-expect";

import React from "react";
import waitForExpect from "wait-for-expect";

import { MockedProvider } from "@apollo/react-testing";
import { act, render } from "@testing-library/react";

import { currentUserQuery } from "../src/graphql/queries";
import LoginPage from "../src/pages/login";

describe("login", () => {
  test("renders correctly", async () => {
    await act(async () => {
      const { getByText } = render(
        <MockedProvider
          mocks={[
            {
              request: {
                query: currentUserQuery,
              },
              result: {
                data: {
                  currentUser: {
                    user: null,
                  },
                },
              },
            },
          ]}
          addTypename={false}
        >
          <LoginPage />
        </MockedProvider>
      );

      await waitForExpect(async () => {
        const LoginButton = getByText("Ingresar");

        expect(LoginButton).toBeTruthy();
        expect(LoginButton).toHaveAttribute("disabled");
      });
    });
  });
});

export {};
