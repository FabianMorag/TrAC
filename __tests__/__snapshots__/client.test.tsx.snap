// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <AdmissionDropout /> Test Snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-xtzu7m"
  >
    <div
      class="css-2tg663"
    >
      <div
        class="chakra-stack unselectable css-11nyuxh"
      >
        <p
          class="chakra-text verticalText css-1qj2vkl"
        >
          Informaci√≥n Complementaria
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
